### ISSUE TITLE
About vgg16.py

### ISSUE THREAD

**Issue by Mickeybaby** [ppengtang/pcl.pytorch#7]
Hi,thank you for your great work!
I want to ask a question.
In class dilated_conv5_body,conv4 and conv5 have no pooling layer.
I agree that conv5 should not have the pooling layer.
But I can't understand why conv4 doesn't have a pool layer.
And according to your code,the vgg16 has only three maxpooling layer,so the spatial_scale should be 1/6.But your code is 1/8.
I don't know if I got it wrong. May I trouble you to give me some advice?
Best wishes！

**Comment by Mickeybaby** [comment]
I'm a beginner. I'm sorry to bother you.I have two other questions. 
No.1  I can't find the SPP Net.
No.2   from .._ext import pcl_losses.I can't open the file,I want to know the function of it.
Best wishes to you！

**Comment by Mickeybaby** [comment]
No.2 from .._ext import pcl_losses
Are you encapsulating the loss function of PCL in it? Namely the section 3.4.3 of the paper ,the formula（8）

**Comment by Mickeybaby** [comment]
Hello, some doubts have been solved by myself.
Now,I'm still confused about this issue.
According to your code,the vgg16 has only three maxpooling layer,so the spatial_scale should be 1/6.But your code is 1/8.
Thanks！

**Comment by ppengtang** [comment]
Hi,

I feel so sorry for the delayed reply.
The answers for some questions are listed as follows.

1. I can't understand why conv4 doesn't have a pool layer.
As we stated in Section 4.1.2, to increase the feature map size from the last conv layer, we replace the penultimate max-pooling layer and its subsequent conv layers by the dilated conv layers.

2. According to your code,the vgg16 has only three maxpooling layer, so the spatial_scale should be 1/6. But your code is 1/8.
The spatial_scale should be (1/2) * (1/2) * (1/2) = 1/8.

3. I can't find the SPP Net.
Please refer to the [RoI pooling layer](https://github.com/ppengtang/pcl.pytorch/tree/master/lib/model/roi_pooling).

4. from .._ext import pcl_losses. Are you encapsulating the loss function of PCL in it? Namely the section 3.4.3 of the paper, the formula (8).
Yes.

Hope these help you.

**Comment by Mickeybaby** [comment]
I got it!
Thank you for your patient guidance.
